src/compiler/compile-props.js

```diff
@@ -229,10 +229,10 @@ function makePropsLinkFn (props) {
 
 function processPropValue (vm, prop, rawValue, fn) {
   const isSimple = prop.dynamic && isSimplePath(prop.parentPath)
-  let value = coerceProp(prop, rawValue)
-  if (value === undefined) {
-    value = getPropDefaultValue(vm, prop)
+  if (rawValue === undefined) {
+    rawValue = getPropDefaultValue(vm, prop)
   }
+  let value = coerceProp(prop, rawValue)
```

Comment:
in most cases you wouldn't use the two together actually... coerce can already take care of default values.